### PR TITLE
add method to get host os/arch in hack scripts

### PR DESCRIPTION
**Pull request by dixudx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
We'd better add a consistent environment in hack scripts. So others can easily port to other architecutres.

**Which issue(s) this PR fixes**:

xref #72804, #72772

**Special notes for your reviewer**:
/cc spiffxp fejta bentheelder

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
hmmm. summoning some more members of the bash gang
/assign @cblecker @dims @ixdy

**Comment by dixudx** [comment]
@lubinsz FYI. Your PRs can rely on this pr. We'd better avoid adding the same logic to fetch the architecutre in different scripts. It's better to keep them consistent.

File: hack/lib/init.sh [hunk 1]
```diff
@@ -40,6 +40,28 @@ export no_proxy=127.0.0.1,localhost
...
+    ;;
+  aarch64*)
+    HOST_ARCH=arm64
+    ;;
+  ppc64le*)
+    HOST_ARCH=ppc64le
+    ;;
+  *)
+    echo "Unsupported Architecture: $(uname -m)."
+    exit 1
```
Comment (BenTheElder) [inline hack/lib/init.sh]: not the most comfortable with init.sh exiting over this..

**Comment by dims** [comment]
@dixudx don't we already have kube::util::host_platform ?

**Comment by dims** [comment]
/hold

we may be duplicating functionality

**Comment by dixudx** [comment]
> don't we already have kube::util::host_platform ?

@dims oh, yes... They are local envs currently. We can export them as gloabl envs.

Updated it.

**Comment by dixudx** [comment]
> Why do we need to export these to the environment?

@cblecker Currently our hack scripts can not run non-amd64 platforms, such as arm64 or ppc64le. That's what #72804 and #72772 are trying to fix.

**Comment by dims** [comment]
@dixudx i believe the question is ... why do we need to export them as global, why can't we have method calls that return the info when/where required.

**Comment by lubinsz** [comment]
ping @dixudx

**Comment by dixudx** [comment]
@dims @BenTheElder Updated. PTAL. Thanks.

**Comment by dims** [comment]
/lgtm

@cblecker can you please release `/hold`?

**Review by BenTheElder — APPROVED** [review]
/lgtm
Thanks

**Comment by ixdy** [comment]
/sig testing
/priority important-longterm
/lgtm
/approve

I updated the PR title and removed the release note because I don't think this really counts as a user-facing change.

**Comment by ixdy** [comment]
(also /sig bash :)

**Review by cblecker — APPROVED** [review]

**Pull request merged**